src/TrajectoryPlanner/src/TrajectoryGenerator.cpp

```diff
@@ -151,6 +152,51 @@ bool TrajectoryGenerator::configurePlanner(const yarp::os::Searchable& config)
     m_dcmGenerator->setFootOriginOffset(leftZMPDelta, rightZMPDelta);
     m_dcmGenerator->setOmega(sqrt(9.81/comHeight));
 
+    // step adjustment
+    std::shared_ptr<UnicyclePlanner> unicyclePlannerStepAdj = m_trajectoryGeneratorStepAdj.unicyclePlanner();
+    ok = ok && unicyclePlannerStepAdj->setDesiredPersonDistance(m_referencePointDistance(0),
+                                                                m_referencePointDistance(1));
+    ok = ok && unicyclePlannerStepAdj->setControllerGain(unicycleGain);
+    ok = ok && unicyclePlannerStepAdj->setMaximumIntegratorStepSize(m_dT);
+    ok = ok && unicyclePlannerStepAdj->setMaxStepLength(maxStepLength);
+    ok = ok && unicyclePlannerStepAdj->setWidthSetting(minWidth, m_nominalWidth);
+    ok = ok && unicyclePlannerStepAdj->setMaxAngleVariation(maxAngleVariation);
+    ok = ok && unicyclePlannerStepAdj->setCostWeights(positionWeight, timeWeight);
+    ok = ok && unicyclePlannerStepAdj->setStepTimings(minStepDuration,
+                                               maxStepDuration, nominalDuration);
+    ok = ok && unicyclePlannerStepAdj->setPlannerPeriod(m_dT);
+    ok = ok && unicyclePlannerStepAdj->setMinimumAngleForNewSteps(minAngleVariation);
+    ok = ok && unicyclePlannerStepAdj->setMinimumStepLength(minStepLength);
+    ok = ok && unicyclePlannerStepAdj->setSlowWhenTurnGain(slowWhenTurningGain);
+    unicyclePlannerStepAdj->addTerminalStep(true);
+    unicyclePlannerStepAdj->startWithLeft(m_swingLeft);
+    unicyclePlannerStepAdj->resetStartingFootIfStill(startWithSameFoot);
+
+    ok = ok && m_trajectoryGeneratorStepAdj.setSwitchOverSwingRatio(switchOverSwingRatio);
+    ok = ok && m_trajectoryGeneratorStepAdj.setTerminalHalfSwitchTime(lastStepSwitchTime);
+    ok = ok && m_trajectoryGeneratorStepAdj.setPauseConditions(maxStepDuration, nominalDuration);
+
+    if (m_useMinimumJerk)
+    {
+        m_feetGeneratorStepAdj = m_trajectoryGeneratorStepAdj.addFeetMinimumJerkGenerator();
+    }
+    else
+    {
+        m_feetGeneratorStepAdj = m_trajectoryGeneratorStepAdj.addFeetCubicSplineGenerator();
+    }
+    ok = ok && m_feetGeneratorStepAdj->setStepHeight(stepHeight);
+    ok = ok && m_feetGeneratorStepAdj->setFootLandingVelocity(landingVelocity);
+    ok = ok && m_feetGeneratorStepAdj->setFootApexTime(apexTime);
+    ok = ok && m_feetGeneratorStepAdj->setPitchDelta(pitchDelta);
+
+    m_heightGeneratorStepAdj = m_trajectoryGeneratorStepAdj.addCoMHeightTrajectoryGenerator();
+    ok = ok && m_heightGeneratorStepAdj->setCoMHeightSettings(comHeight, comHeightDelta);
+    ok = ok && m_trajectoryGeneratorStepAdj.setMergePointRatio(mergePointRatio);
+
+    m_dcmGeneratorStepAdj = m_trajectoryGeneratorStepAdj.addDCMTrajectoryGenerator();
+    m_dcmGeneratorStepAdj->setFootOriginOffset(leftZMPDelta, rightZMPDelta);
+    m_dcmGeneratorStepAdj->setOmega(sqrt(9.81/comHeight));
```

Comment:
Probably all these could have been substituted into two or three lines if there was a copy operator for the UnicycleGenerator. Unfortunately, it is not there :sweat_smile:. The upside of having it would be that we can copy the original generator without having to set again all the settings, thus reducing the risk of errors. Let me know if you are interested in this feature or if you prefer to have the settings of the two objects separated.

Reply:
Actually, this problem would disappear if, instead of duplicating the ``UnicycleGenerator`` object, we duplicate the whole ``TrajectoryGenerator`` in the ``WalkingModule``.

Reply:
Thanks for your explanations so what is your suggestion among the above proposals?

Reply:
I would give it a try in duplicating the ``TrajectoryGenerator`` object. If this gets too complex, simply make it clear for the name of the methods whether you are accessing the original or the adjusted trajectories.

Reply:
Done!